compiler/optimizations/remoteValueForwarding.cpp

```diff
@@ -84,6 +88,83 @@ void remoteValueForwarding() {
   }
 }
 
+static void fixLHSArray(CallExpr* move, std::vector<Symbol*>& todo) {
+  Symbol* LHS = toSymExpr(move->get(1))->symbol();
+  if (LHS->isRef()) {
+    LHS->type = LHS->getValType();
+    LHS->qual = QUAL_VAL;
+    todo.push_back(LHS);
+  }
+}
+
+static void replaceRecordWrappedRefs() {
+  std::vector<Symbol*> todo;
+  forv_Vec(AggregateType, aggType, gAggregateTypes) {
+    if (!aggType->symbol->hasFlag(FLAG_REF)) {
+      for_fields(field, aggType) {
+        if (field->isRef() && isRecordWrappedType(field->getValType())) {
+          field->type = field->getValType();
+          field->qual = QUAL_VAL;
+          todo.push_back(field);
+        }
+      }
+    }
+  }
+
+  // I'd like to be able to just iterate over the uses of tuple fields, but
+  // we don't have a good way of doing that today. The case to worry about
+  // is when we're indexing into a tuple with an integer (param or otherwise).
```

Comment:
We could add another argument to PRIM_GET_SVEC_MEMBER_VALUE that was a SymExpr pointing to the tuple's TypeSymbol. That way, we could use forSymbolSymExprs on the tuple's type symbol to find these.